provider/core/src/serde/mod.rs

```diff
@@ -94,7 +94,7 @@ impl From<postcard::Error> for Error {
     }
 }
 
-#[cfg(feature = "erased-serde")]
+#[cfg(feature = "datagen")]
 impl From<erased_serde::Error> for Error {
```

Comment:
I somewhat have the opposite preference, since

 - we don't test optional dependency features on  the `feature` CI job
 - people will look at the source from rustdoc and notice the feature, and want to use it

so typically i tend to only name features in the source that i actually want people to use directly

Reply:
I'm with Manish, I think implicitely using a dependency as a feature is confusing.

Reply:
> we don't test optional dependency features on the `feature` CI job

This is enough for me, though I'm not convinced by the other argument.